RecoTracker/LSTCore/src/alpaka/NeuralNetwork.h

```diff
@@ -288,6 +289,145 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE::lst {
+
+                                 eta5 - eta4,
+                                 cms::alpakatools::deltaPhi(acc, phi5, phi4),
+                                 (z5 - z4) / dnn::t5dnn::kZ_max,
+                                 (r5 - r4) / dnn::t5dnn::kR_max,
+
+                                 1.0f / innerRadius,
+                                 1.0f / bridgeRadius,
+                                 1.0f / outerRadius,
+
+                                 fakeScore1,
+                                 promptScore1,
+                                 dispScore1,
+                                 (fakeScore2 - fakeScore1),
+                                 (promptScore2 - promptScore1),
+                                 (dispScore2 - dispScore1)};
+
+      float h1[kHiddenFeatures];
+      float h2[kHiddenFeatures];
+
+      linear_layer<kInputFeatures, kHiddenFeatures>(x, h1, dnn::t5embdnn::wgtT_fc1, dnn::t5embdnn::bias_fc1);
+      relu_activation<kHiddenFeatures>(h1);
+
+      linear_layer<kHiddenFeatures, kHiddenFeatures>(h1, h2, dnn::t5embdnn::wgtT_fc2, dnn::t5embdnn::bias_fc2);
+      relu_activation<kHiddenFeatures>(h2);
+
+      linear_layer<kHiddenFeatures, Params_T5::kEmbed>(h2, embedding, dnn::t5embdnn::wgtT_fc3, dnn::t5embdnn::bias_fc3);
+    }
+
+  }  // namespace t5embdnn
+
+  namespace plsembdnn {
+    template <typename TAcc>
+    ALPAKA_FN_ACC ALPAKA_FN_INLINE void runEmbed(TAcc const& acc,
+                                                 const float eta,
+                                                 const float etaErr,
+                                                 const float phi,
+                                                 const float circleCenterX,
+                                                 const float circleCenterY,
+                                                 const float circleRadius,
+                                                 const float ptIn,
+                                                 const float ptErr,
+                                                 const bool isQuad,
+                                                 float (&embedding)[Params_pLS::kEmbed]) {
+      constexpr unsigned int kInputFeatures = 10;
+      constexpr unsigned int kHiddenFeatures = 32;
+
+      float x[kInputFeatures] = {eta / dnn::plsembdnn::kEta_norm,
+                                 etaErr / dnn::plsembdnn::kEtaErr_norm,
+                                 alpaka::math::cos(acc, phi),
+                                 alpaka::math::sin(acc, phi),
+                                 1.0f / ptIn,
+                                 alpaka::math::log10(acc, ptErr),
+                                 isQuad ? 1.0f : 0.0f,
+                                 alpaka::math::log10(acc, alpaka::math::abs(acc, circleCenterX)),
+                                 alpaka::math::log10(acc, alpaka::math::abs(acc, circleCenterY)),
+                                 alpaka::math::log10(acc, circleRadius)};
```

Comment:
I think that this has some insufficient information to connect the momentum direction eta-phi used to define pLS with the position-based eta-phi of the T5 objects.

IIRC, the pLS eta,phi match the outer MD (which is actually the last hit of the pixel track). 
I'd expect that radius and z for that hit are needed to figure out a consistent match with T5.

... also, I recall that the hit/MD details in pLS are stored in a weird way at least partly (like x is actually pt). So, some explicit check that values make sense geometrically would be needed before throwing inputs in the ML.

@alexandertuna 
you may want to add these when you explore the embedding network